velox/type/TimestampConversion.cpp

```diff
@@ -593,6 +633,57 @@ Status daysSinceEpochFromWeekDate(
   return Status::OK();
 }
 
+Expected<int64_t> daysSinceEpochFromWeekOfMonthDate(
+    int32_t year,
+    int32_t month,
+    int32_t weekOfMonth,
+    int32_t dayOfWeek,
+    bool lenient) {
+  if (!lenient &&
+      !isValidWeekOfMonthDate(year, month, weekOfMonth, dayOfWeek)) {
+    if (threadSkipErrorDetails()) {
+      return folly::makeUnexpected(Status::UserError());
+    } else {
+      return folly::makeUnexpected(Status::UserError(
+          "Date out of range: {}-{}-{}-{}",
+          year,
+          month,
+          weekOfMonth,
+          dayOfWeek));
+    }
+  }
+  // If the month is out of range, adjust it into range
+  int32_t additionYears;
+  if (month < 1) {
+    additionYears = month / 12 - 1;
+    month = 12 - abs(month) % 12;
+  } else {
```

Comment:
nit: perhaps use `else if (month > 12)` to avoid unnecessary computing for a valid month.

Reply:
Thank you. This is better. Done.